src/frontend/sharedComponents/PhotoUnpreparedView.tsx

```diff
@@ -0,0 +1,77 @@
+import * as React from 'react';
+import {View, StyleSheet, Image, ActivityIndicator} from 'react-native';
+
+import {AlertIcon} from './icons';
+import type {PhotoVariant, ViewStyleProp} from '../sharedTypes';
+import {useAttachmentUrlQuery} from '../hooks/server/media';
+import {useObservation} from '../hooks/server/observations.ts';
+
+type Props = {
+  observationId: string;
+  attachmentId: string;
+  variant: PhotoVariant;
+  style?: ViewStyleProp;
+  resizeMode?: 'cover' | 'contain' | 'stretch' | 'center';
+};
+
+const PhotoUnpreparedComponent = ({
+  observationId,
+  attachmentId,
+  variant,
+  resizeMode = 'contain',
+  style,
+}: Props) => {
+  const {
+    data: observation,
+    isError: observationError,
+    isLoading: observationLoading,
+    isPending: observationPending,
+  } = useObservation(observationId);
+  const {
+    data: attachmentUrl,
+    isError: attachmentError,
+    isLoading: attachmentUrlLoading,
+  } = useAttachmentUrlQuery(
+    observation.attachments.find(
+      attachment => attachment.driveDiscoveryId === attachmentId,
+    )!,
+    variant,
+    !observationPending,
+  );
+  const isLoading = observationLoading || attachmentUrlLoading;
+  const isError = observationError || attachmentError;
+
+  return (
+    <View style={[styles.container, style]}>
+      {isLoading ? (
+        <ActivityIndicator size="large" />
+      ) : isError ? (
+        <AlertIcon size={96} />
```

Comment:
2 slight problems.
1. When there is an Error, the background is white, so you cannot see the back button. We can fix this by adding `backgroundColor:BLACK` to styles.container
2. Can you replace `ActivityIndicator` with 'UIActivityIndicator' from 'react-native-indicators'. You will need to make it WHITE otherwise you wont see it on the black background